lib/VM/JSLib/TypedArray.cpp

```diff
@@ -1746,6 +1746,126 @@ typedArrayPrototypeToLocaleString(void *, Runtime &runtime, NativeArgs args) {
+  if (LLVM_UNLIKELY(!index.isNumber())) {
+    return false;
+  }
+
+  // 3. If index is -0𝔽, return false.
+  if (LLVM_UNLIKELY(
+          index.getDouble() == 0 && std::signbit(index.getDouble()))) {
+    return false;
+  }
+
+  // 4. If ℝ(index) < 0 or ℝ(index) ≥ O.[[ArrayLength]], return false.
+  if (index.getNumber() < 0 || index.getNumber() >= O->getLength()) {
+    return false;
+  }
+
+  return true;
+}
+
+/// ES14.0 23.2.3.36
+CallResult<HermesValue>
+typedArrayPrototypeWith(void *, Runtime &runtime, NativeArgs args) {
+  // 2. Perform ? ValidateTypedArray(O).
+  if (JSTypedArrayBase::validateTypedArray(runtime, args.getThisHandle()) ==
+      ExecutionStatus::EXCEPTION) {
+    return ExecutionStatus::EXCEPTION;
+  }
+
+  // 1. Let O be this value
+  auto self = args.vmcastThis<JSTypedArrayBase>();
+
+  // 3. Let len be O.[[ArrayLength]].
+  double len = self->getLength();
+
+  // 4. Let relativeIndex be ? ToIntegerOrInfinity(index).
+  auto relativeIndexRes = toIntegerOrInfinity(runtime, args.getArgHandle(0));
+  if (LLVM_UNLIKELY(relativeIndexRes == ExecutionStatus::EXCEPTION)) {
+    return ExecutionStatus::EXCEPTION;
+  }
+
+  // Use double here, because ToInteger may return Infinity.
+  double relativeIndex = relativeIndexRes->getNumber();
+
+  // 5. If relativeIndex ≥ 0, let actualIndex be relativeIndex.
+  // 6. Else, let actualIndex be len + relativeIndex.
+  double actualIndex =
+      convertNegativeBoundsRelativeToLength(relativeIndex, len);
+
+  // 7. If O.[[ContentType]] is BigInt, let numericValueBe ? ToBigInt(value)
+  // 8. Else, let numericValue be ? ToNumber(value)
+  CallResult<HermesValue> res = ExecutionStatus::EXCEPTION;
+  switch (self->getKind()) {
+    default:
+      res = toNumber_RJS(runtime, args.getArgHandle(1));
+      break;
+    case CellKind::BigInt64ArrayKind:
+    case CellKind::BigUint64ArrayKind:
+      res = toBigInt_RJS(runtime, args.getArgHandle(1));
+      break;
+  }
+  if (res == ExecutionStatus::EXCEPTION) {
+    return ExecutionStatus::EXCEPTION;
+  }
+
+  HermesValue actualIndexValue =
+      HermesValue::encodeTrustedNumberValue(actualIndex);
+  // 9. If IsValidIntegerIndex(O, 𝔽(actualIndex)) is false, throw a RangeError
+  // exception.
+  if (LLVM_UNLIKELY(!isValidIntegerIndex(runtime, self, actualIndexValue))) {
+    return runtime.raiseRangeError("index invalid or out of range");
+  }
+
+  // 10. Let A be ? TypedArrayCreateSameType(O, « 𝔽(len) »).
+  auto status = JSTypedArrayBase::allocateSpecies(runtime, self, len);
+  if (LLVM_UNLIKELY(status == ExecutionStatus::EXCEPTION)) {
+    return ExecutionStatus::EXCEPTION;
+  }
+
+  auto A = status.getValue();
+  if (actualIndex > 0) {
+    if (LLVM_UNLIKELY(
+            JSTypedArrayBase::setToCopyOfTypedArray(
+                runtime, A, 0, self, 0, actualIndex) ==
+            ExecutionStatus::EXCEPTION)) {
+      return ExecutionStatus::EXCEPTION;
+    }
+  }
+
+  auto targetValue = runtime.makeHandle(res.getValue());
```

Comment:
This handle needs to be created much earlier - right after `res` is checked for `EXCEPTION`. Otherwise we're potentially holding it across allocations.